### PR TITLE
Add `namedpersistent`

### PR DESCRIPTION
**Pull request by crcrpar**
This PR adds `namedpersistent` method to `chainer.Link`, `chainer.Chain`, and `chainer.ChainList`.

Recent GANs sometimes tweak batch normalizations' moving averages. So, this method will be useful to some extent and pytorch implements a corresponding method.

- [x] write tests: (EDIT)To avoid unprecedented side effect, define almost the same chain in a TestChain.test_namedpersistent.

### CODE REVIEW

**Comment by muupan** [comment]
This feature is also desired for ChainerRL to enable sharing persistent values among multiple processes using shared arrays.

**Comment by kuenishi** [comment]
I marked this as medium priority as this is important for my experiments, although we have temporary implementation for this feature.

**Comment by beam2d** [comment]
bump. @kmaehashi Do you have time to review it?

File: chainer/link.py [hunk 1]
```diff
@@ -1202,6 +1226,15 @@ def namedparams(self, include_uninit=True):
             for path, param in link.namedparams(include_uninit):
                 yield prefix + path, param
 
+    def namedpersistent(self):
+        # type: () -> tp.Iterator[tp.Tuple[str, tp.NdArray]]
+        for ret in super(ChainList, self).namedpersistent():
+            yield ret
+        for idx, link in enumerate(self._children):
+            prefix = '/{}'.format(idx)
```
Comment (toslunar) [inline chainer/link.py]:
```suggestion
            prefix = '/%d' % idx
```
to be consistent with `namedparams` and `namedlinks`.

**Comment by crcrpar** [comment]
The failure came from chainerx tests.

**Comment by kmaehashi** [comment]
As Chainer is under the maintenance phase, unfortunately we decided not to merge this feature. Thanks again for your contribution.
